### DISCUSSION TITLE
Memory utility doesn't seem to drop after a few queries.

### DISCUSSION THREAD

**Discussion by sgf** [ZoneTree/ZoneTree.FullTextSearch#15] [Q&A] [answered]
Memory utility doesn't seem to drop after a few queries.

Memory usage at the time of startup.

![image](https://github.com/user-attachments/assets/66283943-f77b-4a77-b59a-43e0414c76c8)

![image](https://github.com/user-attachments/assets/294e9602-6bfb-4f7b-ba8b-e8e6df50a854)

Memory usage after a few queries.

**Comment by koculu** [comment, marked as answer]
The default block cache life time is 60 seconds: [blockCacheLifeTimeInMilliseconds](https://github.com/koculu/ZoneTree.FullTextSearch/blob/c25884605a7654aaeb55f56d7a8a63a698a1172e/src/ZoneTree.FullTextSearch/SearchEngines/HashedSearchEngine.cs#L53)

After you do a search the data in the disk stays 60 seconds in the memory by default.
After that GC is responsible to release the memory.
You can configure the parameter according to your needs.
